### PR TITLE
Implement Tangram

### LINKED ISSUE #1738: Implement Tangram
https://github.com/broadinstitute/Tangram

https://www.nature.com/articles/s41592-021-01264-7

We can use jax here as it seems Tangram does not use minibatching; therefore, Jax should speed things up for this model on CPU.

### PR DESCRIPTION
**Pull request by adamgayoso**
Fixes #1738 

Example on [this](https://github.com/scverse/scvi-tutorials/pull/80) tutorials PR.

Also fixes an issue with the `to()` to device method after removing `JaxModuleWrapper`

### CODE REVIEW

File: scvi/external/tangram/_model.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+            self.module.to(device)
+            logger.info("Jax module moved to CPU.")
+        tensor_dict = self._get_tensor_dict(device=device)
+        training_plan = JaxTrainingPlan(self.module, **plan_kwargs)
+        module_init = self.module.init(self.module.rngs, tensor_dict)
+        state, params = module_init.pop("params")
+        training_plan.set_train_state(params, state)
+        train_step_fn = JaxTrainingPlan.jit_training_step
+        pbar = track(range(max_epochs), style="tqdm", description="Training")
+        history = pd.DataFrame(index=np.arange(max_epochs), columns=["loss"])
```
Comment (adamgayoso) [inline scvi/external/tangram/_model.py:176]: cc @justjhong it's actually pretty flexible

you can use the training plan just as a shell

File: scvi/data/_manager.py [hunk 2]
```diff
@@ -25,6 +26,23 @@
...
+    check_if_view
+        If True, checks if AnnData is a view.
+    check_fully_paired_mudata
+        If True, checks if MuData is fully paired across mods.
+    """
+
+    check_if_view: bool = True
+    check_fully_paired_mudata: bool = True
+
+
```
Comment (adamgayoso) [inline scvi/data/_manager.py:45]: @justjhong what do you think?

File: docs/release_notes/v0.19.0.md [hunk 3]
```diff
@@ -7,6 +7,7 @@
 -   {class}`~scvi.module.base.LossRecorder` is deprecated. Please substitute with {class}`~scvi.module.base.LossOutput` [#1749]
 -   All training plans require keyword args after the first positional argument [#1749]
 -   {class}`~scvi.module.base.JaxModuleClass` absorbed features from the `JaxModuleWrapper`, rendering the `JaxModuleWrapper` obsolote, so it was removed. [#1751]
+-   Add {class}`scvi.external.Tangram` and {class}`scvi.external.TangramMapper` that implement Tangram for mapping scRNA-seq data to spatial data [#1743].
```
Comment (adamgayoso) [inline docs/release_notes/v0.19.0.md:10]:
```suggestion
-   Add {class}`scvi.external.Tangram` and {class}`scvi.external.tangram.TangramMapper` that implement Tangram for mapping scRNA-seq data to spatial data [#1743].
```

File: docs/release_notes/v0.19.0.md [hunk 4]
```diff
@@ -7,6 +7,7 @@
 -   {class}`~scvi.module.base.LossRecorder` is deprecated. Please substitute with {class}`~scvi.module.base.LossOutput` [#1749]
 -   All training plans require keyword args after the first positional argument [#1749]
 -   {class}`~scvi.module.base.JaxModuleClass` absorbed features from the `JaxModuleWrapper`, rendering the `JaxModuleWrapper` obsolote, so it was removed. [#1751]
```
Comment (adamgayoso) [inline docs/release_notes/v0.19.0.md:9]:
```suggestion
-   {class}`~scvi.module.base.JaxBaseModuleClass` absorbed features from the `JaxModuleWrapper`, rendering the `JaxModuleWrapper` obsolote, so it was removed. [#1751]
```

File: scvi/data/_manager.py (hunk 2 above)
Comment (justjhong) [reply scvi/data/_manager.py:45]: seems fine, the user will never interact with this right?

Comment (adamgayoso) [reply scvi/data/_manager.py:45]: correct

File: scvi/data/_manager.py [hunk 5]
```diff
@@ -39,6 +57,9 @@ class AnnDataManager:
     setup_method_args
         Dictionary describing the model and arguments passed in by the user
         to setup this AnnDataManager.
+    mudata_fully_paired
+        Whether to check that a registered :class:`mudata.MuData` object has fully paired
+        observations across modalities.
```
Comment (adamgayoso) [inline scvi/data/_manager.py:62]:
```suggestion
    validation_checks
        DataClass specifying which global validation checks to run on the data object.
```

**Pull request merged**